packages/model-viewer/src/features/scene-graph/model.ts

```diff
@@ -157,6 +166,43 @@ export class Model implements ModelInterface {
     return this[$materials];
   }
 
+  getMaterialByName(name: string): Material[] {
+    return this[$materials].filter(material => {
+      return material.name === name;
+    });
+  }
+
+  intersectMaterial(screenX: number, screenY: number, camera?: Camera):
+      Material[] {
+    if (camera == null && this[$modelScene] == null) {
+      console.error(
+          'Cannot intersect material, no camera provided or is accessible');
+    }
+    this[$raycaster].setFromCamera(
+        new Vector2(screenX, screenY),
+        camera != null ? camera : this[$modelScene]!.getCamera());
+    const hits = this[$raycaster].intersectObject(this[$threeScene], true);
```

Comment:
Have you looked at the existing `positionAndNormalFromPoint()` method? It would be nice to share some code  here, and make the API similar.